src/timeunit.ts

```diff
@@ -95,6 +135,33 @@ export function convert(unit: TimeUnit, date: Date): Date {
         case TimeUnit.MILLISECONDS:
           result.setMilliseconds(date.getMilliseconds());
           break;
+        case TimeUnit.UTCDAY:
```

Comment:
UTCYEAR will already contain YEAR in the name from `containsTimeUnit` method. 

So I doubt if you really need to add all of these casing.